packages/ember-data/lib/system/relationships/state/has_many.js

```diff
@@ -16,6 +16,25 @@ ManyRelationship.prototype = Ember.create(Relationship.prototype);
 ManyRelationship.prototype.constructor = ManyRelationship;
 ManyRelationship.prototype._super$constructor = Relationship;
 
+ManyRelationship.prototype.setup = function(data) {
+  var key   = this.key,
+      value = data[key];
+
+  if (data.links && data.links[key]) {
+    this.updateLink(data.links[key]);
+  }
+
+  if (this.buildURL) {
```

Comment:
We can then check the flag at https://github.com/arenoir/data/blob/build-hasmany-url/packages/ember-data/lib/system/relationships/state/has_many.js#L188 instead of relying on there being a link